ui/src/app/components/help-sidebar.spec.tsx

```diff
@@ -225,133 +300,127 @@ describe('HelpSidebar', () => {
   });
 
   it('should show delete workspace modal on clicking delete workspace', async () => {
-    const wrapper = await component();
-    wrapper
-      .find({ 'data-test-id': 'workspace-menu-button' })
-      .first()
-      .simulate('click');
-    wrapper
-      .find({ 'data-test-id': 'Delete-menu-item' })
-      .first()
-      .simulate('click');
-    await waitForFakeTimersAndUpdate(wrapper);
-    expect(wrapper.find(ConfirmWorkspaceDeleteModal).exists()).toBeTruthy();
+    component();
+    await user.click(await screen.findByLabelText('Open Actions Menu'));
+
+    await user.click(
+      screen.getByRole('button', {
+        name: /delete/i,
+      })
+    );
+    expect(
+      screen.getByText(/warning — all work in this workspace will be lost\./i)
+    ).toBeInTheDocument();
   });
 
   it('should show workspace share modal on clicking share workspace', async () => {
-    const wrapper = await component();
-    wrapper
-      .find({ 'data-test-id': 'workspace-menu-button' })
-      .first()
-      .simulate('click');
-    wrapper
-      .find({ 'data-test-id': 'Share-menu-item' })
-      .first()
-      .simulate('click');
-    await waitForFakeTimersAndUpdate(wrapper);
-    expect(wrapper.find(MockWorkspaceShare).exists()).toBeTruthy();
+    component();
+    await user.click(await screen.findByLabelText('Open Actions Menu'));
+
+    await user.click(
+      screen.getByRole('button', {
+        name: /share/i,
+      })
+    );
+    expect(screen.getByText(/mock workspace share/i)).toBeInTheDocument();
   });
 
   it('should hide workspace icon if on criteria search page', async () => {
     props = { pageKey: 'cohortBuilder' };
-    const wrapper = await component();
+    component();
+    expect(await screen.findByTestId('sidebar-content')).toBeInTheDocument();
     currentCohortCriteriaStore.next([]);
-    await waitForFakeTimersAndUpdate(wrapper);
 
     expect(
-      wrapper.find({ 'data-test-id': 'workspace-menu-button' }).length
-    ).toBe(0);
-    expect(wrapper.find({ 'data-test-id': 'criteria-count' }).length).toBe(0);
+      screen.queryByLabelText('Open Actions Menu')
+    ).not.toBeInTheDocument();
+    expect(screen.queryByTestId('criteria-count')).not.toBeInTheDocument();
+
     currentCohortCriteriaStore.next([criteria1]);
-    await waitForFakeTimersAndUpdate(wrapper);
-    expect(wrapper.find({ 'data-test-id': 'criteria-count' }).length).toBe(1);
+    expect(await screen.findByTestId('criteria-count')).toBeInTheDocument();
   });
 
   it('should update count if criteria is added', async () => {
     props = { pageKey: 'cohortBuilder' };
-    const wrapper = await component();
+    component();
     currentCohortCriteriaStore.next([criteria1, criteria2]);
-    await waitForFakeTimersAndUpdate(wrapper);
-    expect(
-      wrapper.find({ 'data-test-id': 'criteria-count' }).first().props()
-        .children
-    ).toBe(2);
+    expect(await screen.findByText('2')).toBeInTheDocument();
```

Comment:
correct but incomplete I think - how do we know that this is the right "2" ?

Reply:
Good call. I made this more specific.